### ISSUE TITLE
[kythe] command-line: support stdin filelist and redirections

### ISSUE THREAD

**Issue by fangism** [google/verible#536]
For testing `verible-verilog-kythe-extractor` (which now expects a file-list), sometimes I don't want to create a temporary file list, and just redirect it through stdin.

If you're making some assumptions about file locations relative to the file-list location, we should try to remove those assumptions.

Recall that our [`file::GetContents`](https://cs.opensource.google/verible/verible/+/master:common/util/file_util.h;l=63?q=GetCont&sq=&ss=verible%2Fverible) function interprets `-` as "read-from-stdin", which means there is no location of a file-list.

## Describe what is wrong or missing

I have one file `~/testing/verilog/kythe/function_call_named_args.sv`, and tried the following:

```
$ verible-verilog-kythe-extractor --printkythefacts <(ls ~/testing/verilog/kythe/function_call_named_args.sv)
E1014 13:23:31.161627  195322 indexing_facts_tree_extractor.cc:119] Error while reading file: /dev/fd//usr/local/google/home/fangism/testing/verilog/kythe/function_call_named_args.sv
```

```
$ verible-verilog-kythe-extractor --printkythefacts - <<EOF
> testing/verilog/kythe/function_call_named_args.sv
> EOF
E1014 13:24:39.844734  195495 indexing_facts_tree_extractor.cc:119] Error while reading file: -/testing/verilog/kythe/function_call_named_args.sv
```

```
$ echo testing/verilog/kythe/function_call_named_args.sv | verible-verilog-kythe-extractor --printkythefacts -
E1014 13:25:22.080886  195654 indexing_facts_tree_extractor.cc:119] Error while reading file: -/testing/verilog/kythe/function_call_named_args.sv
```

Basically, `-` and `/dev/fd/*` may need special handling to avoid interpreting them as an actual file-path.
Best would be avoiding any dependence on the input being a file or having some path.

**Comment by fangism** [comment]
This might also be a good time to support a `--filelist-relative-path` option to prefix all _relative_ paths appearing in a filelist.  (This would be separate from `-incdir` which is intended for `` `include ``s.)

**Comment by fangism** [comment]
This is done.
We've added flags `--file_list_path`, `--file_list_root` to decouple the effect of different base/working directories.
I've been able to test single-file extraction with these flags and `<(echo "one-file.sv")` for on-the-fly file lists.
